File: php7/src/BackstagePasses.php
<?php


namespace App;


class BackstagePasses extends GildedRose
{
    private $item;

    public function __construct($item) {
        $this->item = $item;
    }

    public function updateQuality(){
        $this->item->quality += 1;

        if ($this->item->sell_in <= 10) {
            $this->item->quality += 1;
        }

        if ($this->item->sell_in <= 5) {
            $this->item->quality += 1;
        }

        if ($this->item->quality > 50) {
            $this->item->quality = 50;
        }

        if ($this->item->sell_in <= 0) {
            $this->item->quality = 0;
        }

        $this->item->sell_in -= 1;
    }
}

File: Java/src/main/java/com/gildedrose/BackstagePasses.java
package com.gildedrose;

public class BackstagePasses extends StorageItem {
    public BackstagePasses(Item item) {
        super(item);
    }

    @Override
    protected void updateQuality() {
        increaseQuality();

        if (item.sellIn < 11) {
            increaseQuality();
        }

        if (item.sellIn < 6) {
            increaseQuality();
        }
    }

    @Override
    protected void updateExpired() {
        item.quality = 0;
    }
}
